chain/neutrino.go

```diff
@@ -322,8 +322,8 @@ func (s *NeutrinoClient) Rescan(startHash *chainhash.Hash, addrs []btcutil.Addre
 	outPoints map[wire.OutPoint]btcutil.Address) error {
 
 	s.clientMtx.Lock()
-	defer s.clientMtx.Unlock()
```

Comment:
it seems safer to just use a defer here? there are several fields that are protected by a mutex in this method that would now be accessed without one. Is it required that the lock be released during `WaitForShutdown`?

Reply:
Needs to be release when dispatching the RescanFinished notification.

Reply:
okay, then maybe we only do that?

Reply:
Not sure I follow. I don't know if it needs to be released for `WaitForShutdown`, so I'd rather keep the same behavior there. I wasn't a fan of releasing and re-acquiring the lock again for dispatching the notification while keeping the `defer` at the top, so I went for the cleaner (IMO) approach.

Reply:
Never mind, fixed.

Reply:
there's also one other case i think
https://github.com/btcsuite/btcwallet/pull/624/files#diff-3d0490616ade8665ccb58730190699bdR334
`s.rescan` is modified under the lock, but there it accessed w/o it

Reply:
Fixed.